pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
Is it actually a local only image, or truly special?  Should we create
a new category to describe it?

Suppressing the warning seems like it is wrong - we don't want to a
have to check the name here.  We can put flag info on the image that
marks the image as special (using the meta flags)